src/components/FilterableProductTable.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react'
+import SearchBar from './SearchBar'
+import ProductTable from './ProductTable'
+
+export default class FilterableProductTable extends Component {
+    
+    constructor(props) {
+        super(props);
+        this.handleSearch = this.handleSearch.bind(this)
+        this.state = {
+            products : this.props.products
+        }
+    }
+
+    handleSearch(searchString, onlyStocked) {
```

Comment:
I would suggest you to make it more simple to create 2 fucntions. One to handle the search and another to handle the checkbox, like that will be easy to operate and make it work properly.

For example: 

```
handleSearch(searchString, onlyStocked) {
    console.log(searchString);
    let productsResult =
      searchString !== ''
        ? this.state.products.filter((product) => {
            if (onlyStocked) {
              return (
                product.name
                  .toLowerCase()
                  .includes(searchString.toLowerCase()) &&
                product.stocked === true
              );
            } else {
              return product.name
                .toLowerCase()
                .includes(searchString.toLowerCase());
            }
          })
        : this.props.products;

    this.setState({
      products: productsResult,
    });
  }
```

With the code above (doing the solution I gave you before to use the state after it's ready) I can make the list of products show up when you delete all the letters from the search bar, although the checkbox stops working. To make it all work fine I had to do a few changes that I will leave for you.

Nevertheless, it's working almost perfectly :P